### ISSUE TITLE
Remove dependency on obsolete ExoPlayer package

### ISSUE THREAD

**Issue by cybersam** [ArnaudCourbiere/RtspExtractor#1]
The `RtspExtractor` class depends on the `com.google.android.exoplayer.source` package, which was deleted from  ExoPlayer on April 10, 2015 (commit [4c8f9a8](https://github.com/google/ExoPlayer/commit/4c8f9a8c6fb6d831d76ef122bf7f479fdaaf18c3)).

It should be updated to use `com.google.android.exoplayer.FrameworkSampleSource` (which is now also deprecated), or ideally `com.google.android.exoplayer.extractor.ExtractorSampleSource`.

This issue depends on (or may be subsumed by) [Issue 2](https://github.com/ArnaudCourbiere/RtspExtractor/issues/2).

**Comment by ArnaudCourbiere** [comment]
Yeah, my goal was to get it to compile with what I had first. Not sure when I'll have time to get around that. Pull request welcome :)

**Comment by cybersam** [comment]
Actually, reading the Javadoc for `FrameworkSampleSource` more closely, it looks like it may be the right subclass to use, despite its deprecation:

> Playing container formats for which an ExoPlayer extractor does not yet exist (e.g. ogg) is a valid use case of this class.

**Comment by ArnaudCourbiere** [comment]
Cool, will check it out when I get a chance.

**Comment by cybersam** [comment]
On the other hand, the [Customization section of the ExoPlayer Developer Guide](https://google.github.io/ExoPlayer/guide.html#customization) states:

> Extractor – If you need to support a container format not currently supported by the ExoPlayer library, consider implementing a custom Extractor class, which can then be used to together with ExtractorSampleSource to play media of that type.

So, it looks like the ideal solution is to create a custom RTSP `Extractor` that can be used by `ExtractorSampleSource`.

BTW: I need to be able to play back an "rtsps" stream (i.e., RTSP over TLS), so ideally the RTSP `Extractor` should support the "rtsp" and "rtsps". This presumably means that the ffmpeg code should be built with the appropriate SSL/TLS options.

**Comment by ArnaudCourbiere** [comment]
for information, I've built the ffmpeg libs using this repo:

https://github.com/cine-io/android-ffmpeg-with-rtmp

Before looking back into building different flavors of the ffmpeg libs I will update ExoPlayer to the latest version and make a simple demo app.
